### DISCUSSION TITLE
Firmware 3.3.0 and ADB

### DISCUSSION THREAD

**Discussion by Unclezz** [seaky/nspanel_pro_tools_apk#116] [Q&A]
Hello,

I just got a new NSPanel Pro and I am on OS Version 2.2.0. I tried to use adb over wifi and of course it does not work ourt of the box. Menawhile the device wanst to update its fw to 3.3.0. If I do so, will I get locked out from running adb and all the nice stuff for HA?

**Comment by seaky** [comment]
if you opened the device by the official way it will remains opened after the fw upgrade,

if not you have to do it after every fw upgrade.
